### ISSUE TITLE
[Linux] Low FPS and crackling audio with kernel 6.10.4

### ISSUE THREAD

**Issue by staticf0x** [ValveSoftware/Dota-2#2782]
#### Your system information

* [System Information](https://gist.github.com/staticf0x/b0da88ee83a6fd3b72cc4799556286d4)
* Have you checked for system updates?: Yes
* Are you using the latest stable video driver available for your system? Yes
* Have you verified the game files?: Yes
* Launch options: `%command% -sdl_displayindex 0 dota_mouse_window_lock 0 -sdlaudiodriver pulse`

#### Please describe your issue in as much detail as possible:
Since today when I updated my system, my FPS dropped to ~30 from >60 and I hear a crackling sound. The low FPS can be seen both in menu and in game.
Tried without `-sdlaudiodriver pulse`, same issue. Tried restarting PC, no apps opened, still happens. Tried disabling second monitor, still same.
Possible culprits during the system update: kernel 6.9.12 -> 6.10.4, libwayland* 1.22 -> 1.23, mutter 46.3 -> 46.4

#### Steps for reproducing this issue:

Just start DotA2 with a similar setup, it happens every time straight from startup.

**Comment by staticf0x** [comment]
I tried booting up kernel 6.9.12 back and I have 120 FPS like usually. So the problem is kernel 6.10.4 but I don't know why.

**Comment by kisak-valve** [comment]
Hello @staticf0x, this issue should also be reported to your video driver vendor if it hasn't been already.

**Comment by jn64** [comment]
Possibly related:
https://gitlab.freedesktop.org/drm/amd/-/issues/3528

The thread is quite long, you can start around here and try similar steps (disable CSM, enable rebar).
https://gitlab.freedesktop.org/drm/amd/-/issues/3528#note_2517761

**Comment by ghost** [comment]
Some of us managed to solve the performance issues on 6.10.4 by disabling CSM support in the BIOS, as documented in this issue: https://github.com/ValveSoftware/Dota-2/issues/2749

**Comment by staticf0x** [comment]
Thank you, I tried disabling CSM support, but it helps only a little bit:

| Kernel | CSM | FPS |
| ------ |--- | --- |
| 6.10.4 | Enabled | ~30 |
| 6.10.4 | Disabled | ~40 |
| 6.9.12 | - | 120 |

I'll keep an eye on the linked amd gitlab issues.

**Comment by eeceebee** [comment]
> Thank you, I tried disabling CSM support, but it helps only a little bit:
> Kernel 	CSM 	FPS
> 6.10.4 	Enabled 	~30
> 6.10.4 	Disabled 	~40
> 6.9.12 	- 	120
> 
> I'll keep an eye on the linked amd gitlab issues.

This is very similar to what I'm experiencing. Disabling CSM helps a bit, but FPS is still much worse (and less stable) than with the previous kernel.

**Comment by jn64** [comment]
Is Xwayland using a lot of CPU when you experience this low FPS?

There may be a 2nd issue compounding with the CSM/rebar issue.

https://gitlab.freedesktop.org/drm/amd/-/issues/3528#note_2524173
https://gitlab.freedesktop.org/drm/amd/-/issues/3538

**Comment by SniksBjorn** [comment]
I do not believe this is a csm/rebar issue, unless I have misunderstood, the problem in [the linked issue](https://gitlab.freedesktop.org/drm/amd/-/issues/3528) is that csm was disabling rebar. I have rebar disabled due to a crash it caused in RDR2 and since the change to kernal 6.10.x (I face the issue with every version of the 6.10 kernal) I have experienced this drop in frame rate too.

**Comment by SniksBjorn** [comment]
Not sure if this is helpful but will add my experience.
I don't play Dota 2 but do play CS2 and am having the same problem there. I have installed Dota 2 to test both for this comment.
To start with the problem for me is with all 6.10 kernals using xorg or wayland. I previously had rebar disabled due to a crash it caused in RDR2 so I have just turned it back on and it seems to have fixed the low fps.
To me this shows the problem could be independent to CSM disabling rebar and instead be something else causing rebar to be needed for the games to run smoothly.

I don't know how to check if rebar is working but I think it would be good to see if CSM was disabling rebar in previous kernal versions.

**Comment by staticf0x** [comment]
Since linux 6.10.10 the FPS got slightly better up to around 60 FPS, but the performance is still noticeably better on 6.9.x with 120 FPS.

**Comment by ghost** [comment]
I can confirm this issue, and since https://gitlab.freedesktop.org/drm/amd/-/issues/3528#note_2517761 is already closed and people there are saying that the issue is fixed I reiterate - IT IS NOT! I'm getting barely 60fps on 6.11.4 compared to 120fps on 6.9 series... This needs to be looked into by the devs of the source engine for a workaround or at least work from valve to help with the upstream fix... Dota 2 is basically one of the most played games on steam and the situation right now is horrible...

**Comment by baaannnaaannn** [comment]
> I can confirm this issue, and since https://gitlab.freedesktop.org/drm/amd/-/issues/3528#note_2517761 is already closed and people there are saying that the issue is fixed I reiterate - IT IS NOT! I'm getting barely 60fps on 6.11.4 compared to 120fps on 6.9 series... This needs to be looked into by the devs of the source engine for a workaround or at least work from valve to help with the upstream fix... Dota 2 is basically one of the most played games on steam and the situation right now is horrible...

I can confirm,issue still exist but video driver vendors says that its fixed.What the hell....

**Comment by SpiritWinD86** [comment]
this issue is still exist and on gitlab opened a [new one](https://gitlab.freedesktop.org/drm/amd/-/issues/3844), because previous is not fixed (for those who doesn't have a re-bar option in the bios - old systems \ system without uefi), and we need someone's help who can "bisect kernels" (whatever that means...) **https://gitlab.freedesktop.org/drm/amd/-/issues/3844** or at least please [upvote issue there](https://gitlab.freedesktop.org/drm/amd/-/issues/3844). no one is helping there. [previous](https://gitlab.freedesktop.org/drm/amd/-/issues/3528) issue was closed. and it's almost 6 months as it's started with 6.10+ kernels...

**Comment by baaannnaaannn** [comment]
> this issue is still exist and on gitlab opened a [new one](https://gitlab.freedesktop.org/drm/amd/-/issues/3844), because previous is not fixed (for those who doesn't have a re-bar option in the bios - old systems \ system without uefi), and we need someone's help who can "bisect kernels" (whatever that means...) **https://gitlab.freedesktop.org/drm/amd/-/issues/3844** or at least please [upvote issue there](https://gitlab.freedesktop.org/drm/amd/-/issues/3844). no one is helping there. [previous](https://gitlab.freedesktop.org/drm/amd/-/issues/3528) issue was closed. and it's almost 6 months as it's started with 6.10+ kernels...

OP from github
Yeah thats the problem that nobody shouldn't ignore 
I tried to figure out it but cant

**Comment by staticf0x** [comment]
Just letting everybody know, it still doesn't work properly in kernel 6.13.10. 60-75 FPS in demo hero, but drops down to 20 in game.

**Comment by SpiritWinD86** [comment]
> Just letting everybody know, it still doesn't work properly in kernel 6.13.10. 60-75 FPS in demo hero, but drops down to 20 in game.

looks like this fix will be in kernel 6.15 (link on issue [gitlab](https://gitlab.freedesktop.org/drm/amd/-/issues/3844) and tags of mentioned [commit](https://gitlab.freedesktop.org/agd5f/linux/-/commit/a755906fb2b8370c43e91ba437ae1b3e228e8b02) there named "amd-drm-fixes-6.15..."). it's just my assumption.
upd. dev [confirmed](https://gitlab.freedesktop.org/drm/amd/-/issues/3844#note_2894413) .

**Comment by baaannnaaannn** [comment]
> > Just letting everybody know, it still doesn't work properly in kernel 6.13.10. 60-75 FPS in demo hero, but drops down to 20 in game.
> 
> looks like this fix will be in kernel 6.15 (link on issue [gitlab](https://gitlab.freedesktop.org/drm/amd/-/issues/3844) and tags of mentioned [commit](https://gitlab.freedesktop.org/agd5f/linux/-/commit/a755906fb2b8370c43e91ba437ae1b3e228e8b02) there named "amd-drm-fixes-6.15..."). it's just my assumption, devs not confirmed it yet.

Yes

**Comment by staticf0x** [comment]
The [commit](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.14.y&id=4754a071137a7a6afffb9ae6d0429c74fbbf40f2) that (partially) fixes this issue has landed in **6.14.5** and in **6.12.26**.

I have tried 6.14.5 from fedora's updates-testing repo and I now get my usual 120 FPS again. So for me at least, this issue is fixed. But I have zero knowledge about the amdgpu code, so I can't say if it works for everyone.

**Comment by SpiritWinD86** [comment]
yep. today i updated my setup to nobara42 with **kernel 6.14.5**, and now it's finally playable with normal fps, even with high video settings.

**Comment by baaannnaaannn** [comment]
> yep. today i updated my setup to nobara42 with **kernel 6.14.5**, and now it's finally playable with normal fps, even with high video settings.

I updated to 6.14.5 too
If that patch changes are there then its fine

**Comment by kisak-valve** [comment]
Closing per the last several comments.
